### DISCUSSION TITLE
Getting network is down errors via host name, but ip works fine.

### DISCUSSION THREAD

**Discussion by lordmortis** [blinksh/blink#1693] [Support] [answered]
### Checklist

- [x] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues) 
  - #1192 might be related, but I do have local network allowed. 

### Configuration

IOS 16.2
Blink v16.0.4.644 (App Store)

I’ve checked that Blink has local network access and it does. 

### Describe the bug

So I have an internal host running Ubuntu 18.04, and when i try to SSH to it using the hostname, it doesn’t work, if i SSH using the ip it does work. I don’t have any ssh hosts defined.

Example session:
<details> 
  <pre>
blink> ping reliquary.moondine.raganit.com
PING reliquary.moondine.raganit.com (172.16.236.130): 56 data bytes
64 bytes from 172.16.236.130: icmp_seq=0 ttl=64 time=2.829 ms
64 bytes from 172.16.236.130: icmp_seq=1 ttl=64 time=2.944 ms

--- reliquary.moondine.raganit.com ping statistics ---
3 packets transmitted, 2 packets received, 33.3% packet loss
round-trip min/avg/max/stddev = 2.829/2.886/2.944/0.058 ms
blink> ssh reliquary.moondine.raganit.com
Error connecting to reliquary.moondine.raganit.com. connError(msg: "Socket error: Network is down")
blink> ssh 172.16.236.130
Connected to 172.16.236.130
Welcome to Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-193-generic x86_64)
</pre>
</details>

**Comment by Numbski** [comment]
By chance is your device jailbroken?

I only ask because I ran into this myself at one point, and it was because I had an issue in my /etc/hosts file in my local device.

**Comment by Numbski** [comment]
Specifically, I think I had moved my hosts file to /etc/hosts.old and forgot to move it back, causing inconsistent name resolution behavior.

**Comment by lordmortis** [comment]
Nope, and if it helps, this device has never been jailbroken. I’m also pretty sure ssh using host names used to work

**Comment by carloscabanero** [comment, marked as answer]
Sorry you are experiencing this issue. Nothing has changed there so the issue may well be temporary. Ping is solving the address using the usual POSIX C functions, but our ssh implementation uses iOS Sockets, which in theory are mostly the same except for iOS optimizations. If I recall, from my experience debugging this part of the code, I think that may be what is going on. If for some reason a socket would try to connect multiple times without success, at one point it will internally mark that part of the network down until it can figure out how to connect there again.

One possible workaround would be to use a Host from config. We also provide ssh2 command, which if I recall has been using the older libssh2 functions, and that may have more luck.

Sorry again, don’t have a magic wand for this one.

**Reply by lordmortis** [reply]
So `ssh2` works (but it doesn’t use the same keys as ssh), as does adding a config for the host with `ssh`.

what’s the difference between `ssh` and `ssh2`? (My search-fu appears to be weak.)

**Reply by lordmortis** [reply]
When you mentioned the thing about the sockets, I tried turning off iCloud private relay (I turned off the `Limit IP Tracking option` for my wifi network) , and sure enough - that also worked

If i connect via wireguard, that also appears to work fine. Interesting!
